torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1649,6 +1649,23 @@ def wrapped_fn(x):
     return wrapped_fn
 
 
+def np_unary_ufunc_integer_promotion_wrapper_with_astype(fn):
```

Comment:
Better Name?

Reply:
What about changing the previous function to use `astype` instead of `dtype=` instead of creating this new function?

Reply:
With `astype`, the computation still happens in double but we just cast the result to `float` in our case. 
With `dtype`, the computation actually happens in the mentioned dtype as well, so I believe it makes sense to use `dtype` wherever applicable.

What do you think?

Reply:
I meant what if we use astype() BEFORE calling the operation?